src/parser/GenericParser.ts

```diff
@@ -42,69 +42,49 @@ export class GenericParser {
 
-            const fileMetricPromises: Promise<[string, FileMetricResults]>[] = [];
+            const fileMetricPromises: Promise<[SourceFile, FileMetricResults]>[] = [];
 
             if (this.config.parseMetrics) {
-                const metricsParser = new MetricCalculator(this.config);
+                const metricsParser = new MetricCalculator();
 
                 for (const [filePath, parsePromise] of parsePromises) {
-                    fileMetricPromises.push(
-                        metricsParser.calculateMetrics(parsePromise).catch((reason) => {
-                            console.error("Error while parsing file metrics");
-                            console.error(reason);
-                            return [
-                                filePath,
-                                {
-                                    fileType: FileType.Error,
-                                    metricResults: new Map([
-                                        ["ERROR", { metricName: "ERROR", metricValue: -1 }],
-                                    ]),
-                                },
-                            ];
-                        }),
-                    );
+                    fileMetricPromises.push(metricsParser.calculateMetrics(parsePromise));
                 }
             }
 
             // We need to ensure that all trees have been parsed before calculating the coupling metrics,
             // as we need to know the language of the file for that:
-            const filesForCouplingMetrics: ParsedFile[] = [];
             const treeParseResults = await Promise.all(parsePromises.values());
 
-            for (const file of treeParseResults) {
-                // Do not try to parse the syntax tree of the file for the coupling metrics
-                // if that file is unsupported or if parsing already failed once:
-                if (file !== null) {
-                    if (isParsedFile(file)) {
-                        filesForCouplingMetrics.push(file);
-                    } else {
-                        unsupportedFiles.push(file.filePath);
-                    }
+            dlog(" --- " + treeParseResults.length + " files detected", "\n\n");
+
+            const filesForCouplingMetrics: ParsedFile[] = [];
+
+            for (const sourceFile of treeParseResults) {
+                if (isErrorFile(sourceFile)) {
+                    const printPath = formatPrintPath(sourceFile.filePath, this.config);
+                    errorFiles.push(printPath);
+
+                    console.error("Error while parsing the syntax tree for the file " + printPath);
+                    console.error(sourceFile.error);
+                } else if (isParsedFile(sourceFile)) {
+                    // Do only include files for calculating the coupling metrics
+                    // that are supported and have been parsed successfully:
+                    filesForCouplingMetrics.push(sourceFile);
                 }
```

Comment:
I had to change this again because the ErrorFile would not be handled when `this.config.parseMetrics` would evaluate to `false`, as then, there would be no further promise chain.